src/Nethermind/Nethermind.Synchronization.Test/FastSync/StateSyncFeedTestsBase.cs

```diff
@@ -121,9 +124,10 @@ protected ContainerBuilder BuildTestContainerBuilder(DbContext dbContext, int sy
             .AddSingleton<INodeStorage>(dbContext.LocalNodeStorage)
 
             // Use factory function to make it lazy in case test need to replace IBlockTree
+            // Use unique ID per test to avoid cache collisions during parallel execution
             .AddSingleton<IBlockTree>((ctx) => CachedBlockTreeBuilder.BuildCached(
-                $"{nameof(StateSyncFeedTestsBase)}{dbContext.RemoteStateTree.RootHash}{BlockTree.BestSuggestedHeader!.Number}",
-                () => Build.A.BlockTree().WithStateRoot(dbContext.RemoteStateTree.RootHash).OfChainLength((int)BlockTree.BestSuggestedHeader!.Number)))
+                $"{GetType().Name}{dbContext.RemoteStateTree.RootHash}{TestChainLength}{Guid.NewGuid()}",
+                () => Build.A.BlockTree().WithStateRoot(dbContext.RemoteStateTree.RootHash).OfChainLength(TestChainLength)))
 
```

Comment:
```suggestion
                $"{GetType().Name)}{dbContext.RemoteStateTree.RootHash}{BlockTree.BestSuggestedHeader!.Number}",
                () => Build.A.BlockTree().WithStateRoot(dbContext.RemoteStateTree.RootHash).OfChainLength((int)BlockTree.BestSuggestedHeader!.Number)))
```

Something like this, it still caches based on the chain length, but considers difference in types

Reply:
hi @flcl42 ,
I couldn't apply it exactly as written because the static BlockTree field was removed as part of this fix. It was the root cause of the race condition (shared state across parallel test fixtures)

I've also removed the per instance _testInstanceId Guid as you suggested, so caching now works properly across test instances of the same type

Is it looks good now for you?